lib/ab_test.rb

```diff
@@ -60,6 +60,8 @@ def bucket(request:, service_provider:, session:, user:, user_session:)
   def include_in_analytics_event?(event_name)
     if should_log.is_a?(Regexp)
       should_log.match?(event_name)
+    elsif should_log.respond_to?(:include?)
+      should_log.include?(event_name)
     elsif !should_log.nil?
       raise 'Unexpected value used for should_log'
     else
```

Comment:
this is like... teasing me. It's almost a perfect match for `case` syntax which uses `===` and that works on Ranges, Sets, Regexps, Procs... but *not* arrays 😭  

But we can splat them which makes arrays work... so this works!

```ruby
case event_name
when *should_log # splat so that arrays work
  true
else
  false
end
```

Reply:
it wouldn't throw an error for an unexpected value, but that might be ok

Reply:
That feels a _little_ too clever for my liking... in that I have no idea how that works 😅

Reply:
just in case you want an explainer!

<details>
<summary>
case inclusion `===` 
</summary>

- most objects implement `===` as the membership inclusion operator 
- `===` is sometimes called "case equality" because that's what `case/when` calls under the hood

```ruby
literal = '1'
arr = ['1']
set = Set['1']
range = 1..1
regex = /1/
callable = proc { |v| v.to_i == 1 }

yes = '1'

literal === yes
# => true
arr === yes
# => false << but this is worked around with the splat
set === yes
# => true
range === yes
# => true
regex === yes
# => true
callable === yes
# => true

no = '2'

literal === no
# => false
arr === no
# => false
set === no
# => false
range === no
# => false
regex === no
# => false
callable === no
# => false
```
</details>

<details>
<summary>
splatting inside case/when
</summary>

The splat is to work around `===` on an array not doing `#include?`
these two are the exact same
```
one_two = ['1', '2']
case '1'
when '1', '2'
  true
end

case '1'
when *one_two # same as '1', '2'
  true
end
```

and then splatting a single item is like `[*x]` in an array, or it is just the single item
</details>

<details>
<summary>
`Proc#===`
</summary>

`Proc#===` is one of like... a billion obscure ways to call a proc

```ruby
callable = proc { |v| "called with #{v.inspect}" }
=> #<Proc:0x0000000105d5f858 (irb):51>
callable === 1
=> "called with 1"
callable[2]
=> "called with 2"
callable.(3)
=> "called with 3"
```

</details>

Reply:
I wasn't expecting to learn so much obscure Ruby internals today 😄  But I do like how that conveniently adds support for a bunch of different variations like Procs! I still think it's a little arcane to feel comfortable implementing. (Also would leave me a bit uneasy to "unofficially" support procs and other objects without explicit test coverage)